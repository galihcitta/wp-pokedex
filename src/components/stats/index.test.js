import React from "react";
import Stats from "./index";

let element;
let totalStats;
beforeEach(() => {
    const stats = [
        {
            base_stat: 45,
            effort: 0,
            stat: {
                name: "speed",
                url: "https://pokeapi.co/api/v2/stat/6/",
            },
        },
        {
            base_stat: 65,
            effort: 0,
            stat: {
                name: "special-defense",
                url: "https://pokeapi.co/api/v2/stat/5/",
            },
        },
        {
            base_stat: 65,
            effort: 1,
            stat: {
                name: "special-attack",
                url: "https://pokeapi.co/api/v2/stat/4/",
            },
        },
        {
            base_stat: 49,
            effort: 0,
            stat: {
                name: "defense",
                url: "https://pokeapi.co/api/v2/stat/3/",
            },
        },
        {
            base_stat: 49,
            effort: 0,
            stat: {
                name: "attack",
                url: "https://pokeapi.co/api/v2/stat/2/",
            },
        },
        {
            base_stat: 45,
            effort: 0,
            stat: {
                name: "hp",
                url: "https://pokeapi.co/api/v2/stat/1/",
            },
        },
    ];
    totalStats = stats.reduce(
        (value, item) => value + (item.base_stat || 0),
        0,
    );
    element = <Stats data={stats}></Stats>;
});

describe("Stats Component", () => {
    it("renders correct Stats and snapshot element <Stats/>", () => {
        expect(toJson(shallow(element))).toMatchSnapshot();
    });
    it("renders correct Stats and snapshot element <Stats.Loading/>", () => {
        expect(
            toJson(shallow(<Stats.Loading></Stats.Loading>)),
        ).toMatchSnapshot();
    });

    it("check element length <Stats/>", () => {
        expect(
            mount(element).find({ className: "stats__list-row" }).length,
        ).toEqual(6);
    });

    it("check element stats Speed img <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-speed" })
                .find("img")
                .prop("src"),
        ).toEqual("ic-speed.png");
    });

    it("check element stats HP img <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-hp" })
                .find("img")
                .prop("src"),
        ).toEqual("ic-health-point.png");
    });

    it("check element stats Attack img <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-attack" })
                .find("img")
                .prop("src"),
        ).toEqual("ic-attack.png");
    });

    it("check element stats Attack check title name <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-attack" })
                .find("span")
                .text(),
        ).toEqual("Attack");
    });

    it("check element stats Attack check point <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-attack" })
                .find({ className: "stats__list-point" })
                .text(),
        ).toEqual("49");
    });

    it("check element stats Attack check hasc class attack <Stats/>", () => {
        expect(
            mount(element)
                .find({ "test-id": "stats-attack" })
                .find({ className: "stats__list-graph" })
                .children()
                .hasClass("attack"),
        ).toEqual(true);
    });

    it("check element stats Attack width graph <Stats/>", () => {
        const attackPoint = mount(element)
            .find({ "test-id": "stats-attack" })
            .find({ className: "stats__list-point" })
            .text();
        const percentage = `${(attackPoint / totalStats) * 100}%`;
        expect(
            mount(element)
                .find({ "test-id": "stats-attack" })
                .find({ className: "stats__list-graph" })
                .children()
                .props("style").style.width,
        ).toEqual(percentage);
    });
});
