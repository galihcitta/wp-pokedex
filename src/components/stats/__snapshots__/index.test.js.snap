// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stats Component renders correct Stats and snapshot element <Stats.Loading/> 1`] = `
<styled.div>
  <div
    className="stats__container"
  >
    <div
      className="stats__list"
    >
      <div
        className="stats__list-row"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <Skeleton
              circle={false}
              height={30}
              style={Object {}}
              width="100%"
            />
          </div>
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="30px"
            />
          </span>
        </div>
        <div
          className="stats__list-graph"
        />
        <div
          className="stats__list-point"
        >
          <Skeleton
            circle={false}
            height={10}
            style={Object {}}
            width="20px"
          />
        </div>
      </div>
      <div
        className="stats__list-row"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <Skeleton
              circle={false}
              height={30}
              style={Object {}}
              width="100%"
            />
          </div>
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="30px"
            />
          </span>
        </div>
        <div
          className="stats__list-graph"
        />
        <div
          className="stats__list-point"
        >
          <Skeleton
            circle={false}
            height={10}
            style={Object {}}
            width="20px"
          />
        </div>
      </div>
      <div
        className="stats__list-row"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <Skeleton
              circle={false}
              height={30}
              style={Object {}}
              width="100%"
            />
          </div>
          <span>
            <Skeleton
              circle={false}
              height={10}
              style={Object {}}
              width="30px"
            />
          </span>
        </div>
        <div
          className="stats__list-graph"
        />
        <div
          className="stats__list-point"
        >
          <Skeleton
            circle={false}
            height={10}
            style={Object {}}
            width="20px"
          />
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;

exports[`Stats Component renders correct Stats and snapshot element <Stats/> 1`] = `
<styled.div>
  <div
    className="stats__container"
  >
    <div
      className="stats__list"
    >
      <div
        className="stats__list-row"
        key="speed"
        test-id="stats-speed"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-speed.png"
            />
          </div>
          <span>
            Speed
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            speed
                                        "
            style={
              Object {
                "width": "14.150943396226415%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          45
        </div>
      </div>
      <div
        className="stats__list-row"
        key="special-defense"
        test-id="stats-special-defense"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-sp-defense.png"
            />
          </div>
          <span>
            Sp.Def
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            special-defense
                                        "
            style={
              Object {
                "width": "20.440251572327046%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          65
        </div>
      </div>
      <div
        className="stats__list-row"
        key="special-attack"
        test-id="stats-special-attack"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-sp-attack.png"
            />
          </div>
          <span>
            Sp.Atk
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            special-attack
                                        "
            style={
              Object {
                "width": "20.440251572327046%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          65
        </div>
      </div>
      <div
        className="stats__list-row"
        key="defense"
        test-id="stats-defense"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-defense.png"
            />
          </div>
          <span>
            Def
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            defense
                                        "
            style={
              Object {
                "width": "15.40880503144654%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          49
        </div>
      </div>
      <div
        className="stats__list-row"
        key="attack"
        test-id="stats-attack"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-attack.png"
            />
          </div>
          <span>
            Attack
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            attack
                                        "
            style={
              Object {
                "width": "15.40880503144654%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          49
        </div>
      </div>
      <div
        className="stats__list-row"
        key="hp"
        test-id="stats-hp"
      >
        <div
          className="stats__list-title"
        >
          <div
            className="stats__list-icon"
          >
            <img
              alt=""
              src="ic-health-point.png"
            />
          </div>
          <span>
            HP
          </span>
        </div>
        <div
          className="stats__list-graph"
        >
          <div
            className="
                                            stats__list-graph__percent 
                                            hp
                                        "
            style={
              Object {
                "width": "14.150943396226415%",
              }
            }
          />
        </div>
        <div
          className="stats__list-point"
        >
          45
        </div>
      </div>
    </div>
  </div>
</styled.div>
`;
